### DISCUSSION TITLE
Pushing the project to the remote repository

### DISCUSSION THREAD

**Discussion by Sooooc** [community/community#137154] [New to GitHub]
Hello, 
i have this error" error: src refspec master does not match any
error: failed to push some refs to " while trying to push the project to the master branch: "git push -u origin master ". 
And i didn't know how to fix it.

**Comment by davevad93** [comment]
Hi @SouhaChabir ,

Be sure to first add and commit the changes in your local repo before pushing them:

1. ```bash
   git add .
     ```
2. ```bash
   git commit -m "commit"
   ```
3. ```bash
   git push origin master
   ```

**Reply by Sooooc** [reply]
@davevad93 
i think the problem while i run "git push -u origin master", because i had this error when i run it "fatal: 'originmaster' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
"

**Reply by davevad93** [reply]
When your run `git remote -v` is any remote named "`origin`" listed in it?

**Reply by Sooooc** [reply]
yes, there are two remotes named origin

**Reply by davevad93** [reply]
I guess it's something like this, right?

If you use `SSH`:
```bash
origin  git@github.com:USER/REPO.git (fetch)
origin  git@github.com:USER/REPO.git (push)
```
or if you use `HTTPS`:

```bash
origin  https://github.com/USER/REPO.git (fetch)
origin  https://github.com/USER/REPO.git  (push)
```

Also be sure to check the branch by running `git branch --show-current` maybe the problem is that the branch is called `main` and not `master`.

**Reply by Sooooc** [reply]
yes the current branch is "main", and when i run 
/git push -u origin main, i had the same error that i mentioned in the first comment.

**Reply by Sooooc** [reply]
@davevad93

**Reply by davevad93** [reply]
Hey @SouhaChabir , according to the error it looks like you don't have permissions to do `git` operations like `git push`, so my question is, are you the owner of the repo or did you just clone it from another user? If you aren't the owner that would explain why you don't have the permissions to do `git` operations. In case the repo is yours when you cloned it was with `SSH` or `HTTPS`?

One more thing, when you run `git remote -v`, the two remotes start with `git@github.com` or `https://github.com`?

**Reply by Sooooc** [reply]
yes, i'm the owner of the repo. And btw i enter my credentials and still don't know why i can't do operation like push. 
 
The two remotes start with https://github.com

**Reply by davevad93** [reply]
Ok @SouhaChabir, so you cloned it first with `HTTPS`, when it asks for the password, do you enter your `Personal Access Token`?

**Reply by Sooooc** [reply]
yes

**Reply by davevad93** [reply]
That's odd @SouhaChabir , do you have issues pushing with all your repos or just this one in particular?

**Reply by Sooooc** [reply]
@davevad93 I fixed it. I couldn't push my project to the main branch, and i don't know why. 
And i created a master branch then i push the project to this branch.

**Reply by davevad93** [reply]
Cool @SouhaChabir , I'm glad you finally fixed it! 👍

**Reply by ghostinhershell** [reply]
Thanks for hopping in here @davevad93! 🚀

**Comment by tenkhen** [comment]
Make sure you have a correct branch name. It seems like `master` doesn't exist. Did you try `git push -u origin main` instead?

**Reply by Sooooc** [reply]
yes, and i have the same error.
